tutorial/LFRic/example3/example3_driver.f90

```diff
@@ -0,0 +1,220 @@
+  use timestepping_config_mod, &
+                              only : timestep_start, &
+                                     timestep_end
+
+  ! Coordinates
+  use assign_coordinate_field_mod, &
+                              only : assign_coordinate_field
+  ! Algorithms
+  use example3_alg_mod,       only : example3_alg_init, &
+                                     example3_alg_step
+  use diagnostic_alg_mod,     only : diagnostic_alg_init, &
+                                     diagnostic_alg_write
+
+  implicit none
+
+  ! Global and local mesh data types
+  type(global_mesh_type), target  :: global_mesh
+  type(global_mesh_type), pointer :: global_mesh_ptr => null()
+  type(mesh_type), target         :: mesh
+  ! Extrusion
+  type(uniform_extrusion_type), target  :: extrusion
+  type(uniform_extrusion_type), pointer :: extrusion_ptr => null()
+  ! Partition
+  type(partition_type)                      :: partition
+  procedure(partitioner_interface), pointer :: partitioner_ptr => null()
+  ! Function spaces Wchi and W3 for coordinate and perturbation field
+  type(function_space_type), target  :: fs_wchi
+  type(function_space_type), pointer :: fs_wchi_ptr => null()
+  type(function_space_type), target  :: fs_w3
+  type(function_space_type), pointer :: fs_w3_ptr => null()
+  ! Coordinate fields
+  type(field_type)                   :: chi(3)
+  ! Perturbation field
+  type(field_type)                   :: perturbation
+  ! Number of data per degree of freedom in fields
+  integer(kind=i_def), parameter     :: ndata_sz = 1
+  ! Maximum depth (of cells outside the cell over which the stencil is based)
+  ! of the stencil to be used on fields with this partition.
+  ! A single cell stencil will, therefore, have a  max_stencil_depth=0.
+  ! A nine-point square region stencil will have max_stencil_depth=1.
+  integer(kind=i_def)                :: max_stencil_depth
+  ! Number of the MPI rank of this process
+  integer(kind=i_native)             :: local_rank
+  ! Total number of MPI ranks (processes) in this job
+  integer(kind=i_def)                :: total_ranks
+  ! Auxiliary variables for coordinate fields (function space ID, loop counters)
+  integer(kind=i_def)                :: chi_space, i, tstep
+  ! Auxiliary variable for naming coordinate fields
+  character(len=str_short)           :: cind
+
+  !-----------------------------------------------------------------------------
+  ! Set model parameters from the configuration file
+  !-----------------------------------------------------------------------------
+  call log_event( "Setting 'example3_driver' model parameters", LOG_LEVEL_INFO )
+  call read_configuration( "configuration.nml", local_rank )
+
+  !-----------------------------------------------------------------------------
+  ! Set partitioner parameters (planar mesh on a single process)
+  !-----------------------------------------------------------------------------
+  max_stencil_depth = 0
+  local_rank = 0
```

Comment:
This line must be moved to before the read_configuration call in line 125, otherwise local_rank is undefined when the configuration is read.

Reply:
Corrected, thanks for spotting! I wondered why it didn't fall apart and printing the value revealed that `gfortran` initialised it to 0. Compilers can be too helpful :worried:

Reply:
My gfortran didn't - that's how I noticed  :)